pkg/prometheus/promcfg.go

```diff
@@ -4104,6 +4114,24 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	return cfg, nil
 }
 
+func (cg *ConfigGenerator) generateOTLPConfig(_ *assets.StoreBuilder) yaml.MapItem {
+	cfg := yaml.MapSlice{}
+
+	otlpConfig := cg.prom.GetCommonPrometheusFields().OTLP
+
+	if len(otlpConfig.PromoteResourceAttributes) > 0 {
+		cfg = append(cfg, yaml.MapItem{
+			Key:   "promote_resource_attributes",
+			Value: otlpConfig.PromoteResourceAttributes,
+		})
+	}
+
+	return yaml.MapItem{
+		Key:   "otlp",
+		Value: cfg,
```

Comment:
We could even go a step further and do the following inside an `if` block.

```suggestion
		Value: yaml.MapItem{
		    Key:   "promote_resource_attributes",
		    Value: otlpConfig.PromoteResourceAttributes,
		    },
```